### ISSUE TITLE
CRUD Delete for list node with identityref key fails

### ISSUE THREAD

**Issue by abhikeshav** [CiscoDevNet/ydk-gen#257]
From Xiaoqin:
> Bug comes from “_match_list_key” function in “_provider_plugin.py”. If the key for list is an Identity type, this function is not able to return correct target node, please change this function to:

```python
 def _match_list_key(self, root, entity):
        target_root = None
        chs = root.getchildren()
        keys = entity.i_meta.key_members()
        for key in keys:
            key_value = getattr(entity, key.presentation_name)
            if key.mtype == REFERENCE_ENUM_CLASS:
                key_value = key_value.name.replace('_', '-').lower()
            elif key.mtype == REFERENCE_IDENTITY_CLASS:
                identity_inst = getattr(entity, key.presentation_name)
                key_value = 'idx:%s' % identity_inst._meta_info().yang_name
            key_value = str(key_value)
            for ch in chs:
                if key.name == ch.tag and key_value == ch.text:
                    target_root = root
                    return target_root
        return target_root
```

**Comment by abhikeshav** [comment]
To reproduce,

```
    crud = CRUDService()
    ncc = NetconfServiceProvider(address='127.0.0.1' , username='admin', password='admin', port=12022)

    b = openconfig_bgp.Bgp()

    a1 = openconfig_bgp.Bgp.Global.AfiSafis.AfiSafi()
    a1.afi_safi_name = openconfig_bgp_types.Ipv4_UnicastIdentity()
    a1.config.afi_safi_name = openconfig_bgp_types.Ipv4_UnicastIdentity()

    a2 = openconfig_bgp.Bgp.Global.AfiSafis.AfiSafi()
    a2.afi_safi_name = openconfig_bgp_types.Ipv6_UnicastIdentity()
    a2.config.afi_safi_name = openconfig_bgp_types.Ipv6_UnicastIdentity()

    b.global_.afi_safis.afi_safi.append(a1)
    b.global_.afi_safis.afi_safi.append(a2)
    
    crud.delete(ncc, b.global_.afi_safis.afi_safi)
```

**Comment by abhikeshav** [comment]
Use this profile to generate bundle: https://github.com/manradhaCisco/ydk-gen/blob/ydk_core/profiles/test/ydktest-cpp.json

**Comment by abhikeshav** [comment]
Fixed in 687d2cd5b2d42f954c97881e61ef52a3d0cf88d6
